numba/tests/test_ssa.py

```diff
@@ -281,3 +281,56 @@ def foo(q):
             return lin[0]
 
         self.check_func(foo, np.zeros((2, 2)))
+
+    def test_issue5482_missing_variable_init(self):
+        # Test error that lowering fails because variable is missing
+        # a definition before use.
+        @njit("(intp, intp, intp)")
```

Comment:
Should probably test in object mode too, so `PyLowering` get exercised?

Reply:
Oh true. I'll add a test.